### PR TITLE
More helpful ami display

### PR DESCRIPTION
**Pull request by adamnfish**
Displays instance AMIs by stack/app/stage to show their usage.

Note: This change drops support for showing AMIs that could not be looked up. If that's required it'll need to be added in another PR. One change at a time!

ssa-amis

### CODE REVIEW

File: test/util/Fixtures.scala [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+import models._
+import org.joda.time.DateTime
 
 object Fixtures {
+  def emptyAmi(arn: String): AMI =
+    AMI(arn, None, "", "", None, Map.empty, None, "", "", "", "", "",  None)
+  def emptyInstance(arn: String): Instance =
+    Instance(arn, "", "", "", "", "", DateTime.now, "", "", "", Nil, Map.empty, None, None, Nil, Nil, Map.empty, Meta("", Origin("", "", "", "")))
+  def instanceWithAmiArn(arn: String, amiArnOpt: Option[String]): Instance =
+    amiArnOpt.fold(emptyInstance(arn))(amiArn => emptyInstance(arn).copy(specification = Map("imageArn" -> amiArn)))
```
Comment (sihil) [inline test/util/Fixtures.scala]: Ha. Somehow I'd never cracked the functional way to do this. Nice pattern.

**Comment by sihil** [comment]
In my mind this would be the other way round and show an AMI with the list of SSAs that are associated with it. Also I think that we should adopt a concise format for the SSA, something like stack:app/stage.

**Comment by adamnfish** [comment]
Soz, merged this for the demo. I misagree because you search by SSA, not by AMI. Let's chat IRL though.

**Comment by sihil** [comment]
Yeah - I'm not particularly attached to it and I think this makes sense. Although I still think we should have a consistent SSA format.

**Pull request merged**